pax-std/pax-std-primitives/src/lib.rs

```diff
@@ -15,10 +15,17 @@ fn patch_if_needed<T: PartialEq + Clone>(
     patch: &mut Option<T>,
     new_value: T,
 ) -> bool {
-    let no_update = old_state.as_ref().is_some_and(|c| &new_value == c);
-    if !no_update {
-        *patch = Some(new_value.clone());
-        *old_state = Some(new_value);
+    optional_patch_if_needed(old_state, patch, Some(new_value))
+}
+
+fn optional_patch_if_needed<T: PartialEq + Clone>(
```

Comment:
is there a reason we create this function when it's only used once? Seems like a pretty short snippet of code.

Reply:
Split them when I wanted to use the second function to send an optional width/height (as described above) and didn't revert. When I do these kind of changes that end up not being used, is it generally preffered to undo them?